src/foam/u2/crunch/wizardflow/LoadCapabilitiesAgent.js

```diff
@@ -41,19 +41,28 @@ foam.CLASS({
       factory: function () {
         return this.WAOSetting.UCJ;
       }
+    },
+    {
+      name: 'subject',
+      class: 'FObjectProperty',
+      of: 'foam.nanos.auth.Subject'
     }
   ],
 
   methods: [
     // If Property expressions ever unwrap promises this method can be blank.
     function execute() {
+      if ( this.subject ) {
+        return this.crunchService.getCapabilityPathFor(null, this.rootCapability.id, false, this.subject.user, this.subject.realUser)
+          .then(capabilities => { this.capabilities = capabilities });
+      }
       return this.crunchService.getCapabilityPath(null, this.rootCapability.id, false)
         .then(capabilities => { this.capabilities = capabilities });
     },
     function getWAO() {
       switch ( this.waoSetting ) {
         case this.WAOSetting.UCJ:
-          return this.UserCapabilityJunctionWAO.create({}, this.__context__);
+          return this.UserCapabilityJunctionWAO.create({ subject: this.subject ? this.subject : null }, this.__context__);
```

Comment:
is this really necessary. If subject is falsy shouldn't UserCapabilityJunctionWAO know that - does it really have to pass back a null?

Reply:
Addressed.